format-common/src/main/kotlin/dev/msfjarvis/aps/data/passfile/PasswordEntry.kt

```diff
@@ -52,13 +57,13 @@ constructor(
    * does not have a TOTP secret, the flow will never emit. Users should call [hasTotp] before
    * collection to check if it is valid to collect this [Flow].
    */
-  public val totp: Flow<String> = flow {
+  public val totp: Flow<Totp> = flow {
     if (totpSecret != null) {
-      repeat(Int.MAX_VALUE) {
-        val (otp, remainingTime) = calculateTotp()
+      do {
+        val otp = calculateTotp()
         emit(otp)
-        delay(remainingTime)
-      }
+        delay(1000L)
```

Comment:
Technically this should be `1000L - Time(calculateTotp())`, right? I don't know how big of a difference would it be accumulated over 60s but we can probably ignore if it is not much.